### DISCUSSION TITLE
Node.js version 12.x–17.x required but found 21.3.0

### DISCUSSION THREAD

**Discussion by tsbertalan** [community/community#102144] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

How can I get copilot working in neovim after the error in the response? 

`which node` gives:
/home/tsbertalan/.nvm/versions/node/v21.3.0/bin/node

So I do `nvm install 17` and `:Copilot setup` works fine.

However, does this mean that whether copilot will work in the future depends on which version of node I happen to have activated in my terminal?? I'm a Python dev, not node, so I'm more used to virtualenvs and giving explicit python executable path for distinguishing between versions.

**Comment by tpope** [comment, marked as answer]
You're on an old version of the plugin. Recent versions support Node 16+.

**Reply by tsbertalan** [reply]
I did

```bash
cd ~/.config/nvim/pack/github/start/copilot.vim/
git pull
```

(which gave `Updating 22fd954..9484e35` among other outputs), and then, in nvim, `:Copilot setup`, but got back `Copilot: 'Node.js version 12.x–17.x required but found 21.3.0'`. Is this the right way to update the plugin in nvim?

**Reply by tpope** [reply]
You probably have 2 installations.

**Reply by tsbertalan** [reply]
You're right. For future reference:

```bash
$ find ~ -name "tree-sitter.wasm" 2>/dev/null
/home/tsbertalan/.config/nvim/pack/github/start/copilot.vim/dist/tree-sitter.wasm
/home/tsbertalan/.vscode-server/extensions/github.copilot-chat-0.13.1/dist/tree-sitter.wasm
/home/tsbertalan/.vscode-server/extensions/github.copilot-1.176.0/dist/tree-sitter.wasm
/home/tsbertalan/.local/share/nvim/plugged/copilot.vim/dist/tree-sitter.wasm
$ cd /home/tsbertalan/.local/share/nvim/plugged/copilot.vim/
$ git pull
$ vim
> :Copilot setup
> :Copilot version
copilot.vim 1.26.0                                                                                                                                                                                                                   
Neovim 0.7.2
GitHub Copilot Language Server 1.173.0
Node.js 21.3.0
Linux
```

Not sure how I got both the `~/.config/` and `~/.local` versions, but whatever, it works now.
